replica/database.cc

```diff
@@ -1032,11 +1025,28 @@ future<> database::drop_column_family(const sstring& ks_name, const sstring& cf_
     f.get(); // re-throw exception from truncate() if any
 }
 
+future<std::vector<foreign_ptr<lw_shared_ptr<table>>>> database::get_table_on_all_shards(sharded<database>& sharded_db, utils::UUID uuid) {
```

Comment:
We want to extract the `lw_shared_ptr<table>` instances before they get dropped by `drop_table_on_all_shards`.
By the time we get to snapshot and truncate the table on each shard the instances are not listed anymore.

Reply:
> We want to extract the `lw_shared_ptr<table>` instances before they get dropped by `drop_table_on_all_shards`. By the time we get to snapshot and truncate the table on each shard the instances are not listed anymore.

But if we lookup the table before drop_column_family() is called (as today), doesn't it work?

dropping a table is always the last step.

Reply:
Now I understand it, it's a preexisting requirement. That's why drop procedure has to keep a shared ptr to table.

Reply:
right.  We do look it up on each shard using either uuid or ks/cf names, and keep the shared pointer to be used later on, after the table shard-instances get unlisted.

As a follow up, I think we can change the design so to have sharded<keyspace> and shard<table> instances (deriving form peering_sharded_service), so we can get to all shard instances using the local instance, since neither are really independent. I.e. you can have a keyspace or a table instantiated not on all shards.